src/plot_api/helpers.js

```diff
@@ -196,6 +229,30 @@ function cleanAxRef(container, attr) {
     }
 }
 
+function cleanTitle(titleContainer) {
+    if(titleContainer) {
+
+        // title -> title.text
+        if(typeof titleContainer.title === 'string') {
```

Comment:
We'll need to be less stringent here. Even though `layout.title` used to be `valType: 'string'`, we also access numbers:

https://github.com/plotly/plotly.js/blob/11bec79d010a44694bf70b357bd11e11ae874ccb/src/lib/coerce.js#L120-L128

For example,

```js
Plotly.newPlot(gd, [], {title: 100, xaxis: {title: 0}, yaxis: {title: 1}})
```

gives:

![image](https://user-images.githubusercontent.com/6675409/48864533-d7702e80-ed9a-11e8-8cf5-b477637e06e2.png)

-------


So

```
if('title' in titleContainer) { // ...
```

should be ok.

Reply:
Ah, I wasn't aware of that. But it's certainly good behavior!

`if('title' in titleContainer) { // ...` would be `true` when `title` is an `object` as well.
So I think 
`if(typeof titleContainer.title === 'string' || typeof titleContainer.title === 'number') { // ...`
will do the job.

Reply:
Well done. Thanks!